aerleon/utils/config.py

```diff
@@ -19,55 +21,49 @@
     'exp_info': 2,
 }
 
+default_file = './aerleon.yml'
 
-def yaml_loader(filename):
-    with open(filename, 'r') as f:
-        try:
-            data = yaml.safe_load(f)
-        except AttributeError:
-            data = yaml.safe_load(f)
-
-    return data
 
+def load_config(
+    config_file: "str | pathlib.Path | list[str | pathlib.Path]" = None,
+    apply_defaults: bool = True,
+) -> dict:
+    """Load Aerleon configuration file(s).
 
-def flags_to_dict(absl_flags):
-    base = {
-        'base_directory': absl_flags.base_directory,
-        'definitions_directory': absl_flags.definitions_directory,
-        'policy_file': absl_flags.policy_file,
-        'output_directory': absl_flags.output_directory,
-        'optimize': absl_flags.optimize,
-        'recursive': absl_flags.recursive,
-        'debug': absl_flags.debug,
-        'verbose': absl_flags.verbose,
-        'ignore_directories': absl_flags.ignore_directories,
-        'max_renderers': absl_flags.max_renderers,
-        'shade_check': absl_flags.shade_check,
-        'exp_info': absl_flags.exp_info,
-    }
+    Args:
+        config_file: An optional string or pathlib.Path with the location of the config file
+            to open. A list of config files can be given. Default is './aerleon.yml'.
+        apply_defaults: Whether to set missing config fields to their default values. Default
+            is True.
 
-    return {flag: base[flag] for flag in filter(lambda f: base[f] is not None, base)}
+    Raises:
+        IOError: If open(config_file) raises an IOError. Note that it is not an error for the
```

Comment:
Don't you mean FileNotFoundError?

Reply:
No, the docstring is correct. FileNotFoundError is handed in this function on line 66. The desired behavior is for load_config to not raise an error if there is not config file. IOError can also be raised by `open()` if there is some more subtle issue with the filesystem and in that case it is up to the caller to decide what to do.

From the docstring on line 44/45:

```
Returns: a dictionary containing the contents of the config file. If no config file is found
        all default values will be returned (unless apply_defaults is False).
```

Reply:
Also from lines 40/41:

```
Note that it is not an error for the config file to not exist.
```